cmd/agent/app/flags.go

```diff
@@ -22,69 +22,83 @@ package app
 
 import (
 	"flag"
-	"fmt"
 	"strings"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	suffixWorkers             = "workers"
+	suffixServerQueueSize     = "server-queue-size"
+	suffixServerMaxPacketSize = "server-max-packet-size"
+	suffixServerHostPort      = "server-host-port"
+	collectorHostPort         = "collector.host-port"
+	httpServerHostPort        = "http-server.host-port"
+	discoveryMinPeers         = "discovery.min-peers"
 )
 
-// Bind binds the agent builder to command line options
-func (b *Builder) Bind(flags *flag.FlagSet) {
+// CreateFlags for this builder based on the application's parameters
```

Comment:
Comment adjusted.

I haven't touched https://github.com/uber/jaeger/blob/master/cmd/agent/app/builder_test.go#L70 and it still works as before.

Creating FlagSet depends on Builder - number of processors.
I will explain:
1. initilize builder
2. create flags for given builder. This has to be after builder creation because the number of flags depends on the number of processors.
3. Init from viper. Default values will resolve to the ones specified at `1.` (except `discoveryMinPeers` which is hardcoded in `CreateFlags`, we can move this to the `NewBuilder` function)